### ISSUE TITLE
cant use this library in react@18.2.0

### ISSUE THREAD

**Issue by galmalka21** [JodusNodus/react-qr-reader#340]
are you going to update the library soon to be able to run in react@18.2.0?

**Comment by Rehankhalil462** [comment]
@galmalka21 use `--legacy-peer-deps` flag while installing this package. It will work with react@18.2.0.

**Comment by dcastillogi** [comment]
For those who are attempting to deploy to Vercel, keep in mind the following steps: Go to _Settings > General > Build & Development Settings_. Then, override the Install Command by using either `npm install --force` or `npm install --legacy-peer-deps`.

**Comment by val-samonte** [comment]
Big issue tbh. If someone created a fork to fix this please drop it here. Thanks.

**Comment by vbsh123** [comment]
If this is still relevant, there's a similiar version that works (idk if its forked from this one though but it is based on this)
https://github.com/kybarg/react-qr-scanner

**Comment by vairaselvam** [comment]
Even I need a solution. I migrated ReactJs from 16 to 18. Now react-qr-reader is throwing error.

**Comment by kelvinCJJ** [comment]
Any solutions?

**Comment by hholmes6** [comment]
same

**Comment by phanthaiduong22** [comment]
I have same issue

**Comment by Jcoloma13** [comment]
"overrides": {
  "react-qr-reader": {
    "react": "$react",
    "react-dom": "$react-dom"
  }
}  

You can try this.

**Comment by adlarionov** [comment]
Just run `npm i --legacy-peer-deps react-qr-reader`, then in your generated `package-lock.json` rewrite `peerDependencies` in `node_mobules/react-qr-reader` field. Lastly, run ` npm i`. Everything should work.
<img width="593" alt="image" src="https://github.com/JodusNodus/react-qr-reader/assets/55808030/733794e8-3b1e-47a0-a23d-42667191127a">

**Comment by h0jeZvgoxFepBQ2C** [comment]
> Just run `npm i --legacy-peer-deps react-qr-reader`, then in your generated `package-lock.json` rewrite `peerDependencies` in `node_mobules/react-qr-reader` field. Lastly, run ` npm i`. Everything should work. <img alt="image" width="593" src="https://user-images.githubusercontent.com/55808030/263995516-733794e8-3b1e-47a0-a23d-42667191127a.png">

I'm not sure if i'm totally right, but just ignoring dependency issues is not the best way forward. this issue should be fixed

**Comment by adlarionov** [comment]
> > Just run `npm i --legacy-peer-deps react-qr-reader`, then in your generated `package-lock.json` rewrite `peerDependencies` in `node_mobules/react-qr-reader` field. Lastly, run ` npm i`. Everything should work. <img alt="image" width="593" src="https://user-images.githubusercontent.com/55808030/263995516-733794e8-3b1e-47a0-a23d-42667191127a.png">
> 
> I'm not sure if i'm totally right, but just ignoring dependency issues is not the best way forward. this issue should be fixed

Yes, you are absolutly right. I just showed my solution to this problem. It is an issue that should be fixed for sure

**Comment by mh-jsx** [comment]
if you have react > 17 Use this instead
https://www.npmjs.com/package/react-zxing

Both the libraries are using @zxing/library under the hood.
